routes/dao_coin_exchange_with_fees.go

```diff
@@ -178,33 +196,70 @@ func (fes *APIServer) UpdateDaoCoinMarketFees(ww http.ResponseWriter, req *http.
 	additionalExtraData := make(map[string][]byte)
 	additionalExtraData[lib.TokenTradingFeesByPkidMapKey] = feeMapByPkidBytes
 
+	// Setting this byte makes it impossible to update the trading fee map in the
+	// future. Use with caution.
+	if requestData.DisableTradingFeeUpdate {
+		SetExtraDataValueTrue(additionalExtraData, lib.DisableTradingFeeUpdateKey)
+	}
+
+	ammPublicKeyBytes, _, err := lib.Base58CheckDecode(fes.Config.AmmMetadataPublicKey)
+	if err != nil || len(ammPublicKeyBytes) != btcec.PubKeyBytesLenCompressed {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"UpdateDaoCoinMarketFees: Problem decoding public key %s: %v",
+			fes.Config.AmmMetadataPublicKey, err))
+		return
+	}
+
+	// Fetch the existing fee association to check if fee updates are still allowed.
+	associationFilter := &lib.UserAssociationQuery{
+		AssociationType:  []byte(lib.TokenTradingFeesByPkidMapKey),
+		AssociationValue: []byte(lib.TokenTradingFeesByPkidMapKey),
+		TransactorPKID:   lib.NewPKID(ammPublicKeyBytes),
+		TargetUserPKID:   lib.NewPKID(profilePublicKeyBytes),
+		AppPKID:          lib.NewPKID(ammPublicKeyBytes),
+	}
+	associations, err := utxoView.GetUserAssociationsByAttributes(associationFilter)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("UpdateDaoCoinMarketFees: Problem fetching associations: %v", err))
+		return
+	}
+	if len(associations) == 1 {
+		association := associations[0]
+		if IsExtraDataValueTrue(association.ExtraData, lib.DisableTradingFeeUpdateKey) {
+			_AddBadRequestError(ww, fmt.Sprintf(
+				"UpdateDaoCoinMarketFees: Trading fee updates are disabled for this market"))
+			return
+		}
+	} else if len(associations) != 0 {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"UpdateDaoCoinMarketFees: Expected at most one association but got %d", len(associations)))
+		return
+	}
```

Comment:
Can we write it like:
1. `if len(associations) > 1`, throw error
2. `if len(associations) == 1`, check extra data value true thing.

I prefer to get rid of else ifs if we can.